### PR TITLE
Object Oriented JS

### PR DESCRIPTION
**Pull request by mazvira**
### Object Oriented JS

[Demo](https://mazvira.github.io/frontend-nanodegree-arcade-game) |
[Code base](https://github.com/mazvira/frontend-nanodegree-arcade-game)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/mazvira/task_JS_OOP/app.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
+const START_X = 200;
+const START_Y = 390;
+const WINDOW_WIDTH = 505
+const STEP_X = 100
+const STEP_Y = 80
```
Comment (OleksiyRudenko) [inline submissions/mazvira/task_JS_OOP/app.js]: Please, be consistent in your syntax. I would stick to having semicolons at the end of every statement.
Besides having single const statement with comma separated definitions e.g.
```
const A = 100,
  B = "fghjk",
  C = {
    propA: "text",
  },
  D = whateverVar;
```
is legal and convenient.

File: submissions/mazvira/task_JS_OOP/app.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    else
+        this.x += dt*this.speed;
+
+    if (player.x < this.x + STEP_X/2 &&
+        player.x + STEP_X/2 > this.x &&
+        player.y < this.y + STEP_Y/4 &&
+        player.y + STEP_Y/4 > this.y) {
+        player.x = START_X;
+        player.y = START_Y;
+    }
```
Comment (OleksiyRudenko) [inline submissions/mazvira/task_JS_OOP/app.js]: `.update` does two important and very different jobs: (a) updates character location and (b) checks if there any collisions.
To make code Separation of Concerns principle compliant, please, implement a dedicated method t check for collisions.

Comment (OleksiyRudenko) [inline submissions/mazvira/task_JS_OOP/app.js]: Classes should be agnostic of the outer world, so referring to globals is counter-OOP (exceptions are global constants and library functions).
`player` is a global variable.
Please, fix.
If you have any questions or need clarifications, please, let me know.

**Comment by mazvira** [comment]
Updated js-oop

**Pull request merged**